packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -225,6 +236,57 @@ export class ExtensionInstance<TConfiguration extends BaseConfigType = BaseConfi
   get isJavaScript() {
     return Boolean(this.entrySourceFilePath?.endsWith('.js') || this.entrySourceFilePath?.endsWith('.ts'))
   }
+
+  get functionExtension(): FunctionExtension | undefined {
+    if (!this.isFunctionExtension) return undefined
+    return this as unknown as FunctionExtension
+  }
+
+  async buildStep(options: ExtensionBuildOptions) {
+    if (this.isThemeExtension) {
+      return buildThemeExtension(this, options)
+    } else if (this.isFunctionExtension) {
+      return buildFunctionExtension(this.functionExtension!, options)
+    } else if (this.features.includes('esbuild')) {
```

Comment:
Breaking the pattern a bit here. Should we rename the function below to `buildExtensionWithEsbuild` or something?

Reply:
I agree actually, `buildUIExtension` does make sense if looked from that angle 👍